src/game.cpp

```diff
@@ -13116,53 +13161,116 @@ void game::climb_down( const tripoint &examp )
-                               "Looks like %d stories.  Nothing your webs can't handle.  Descend?", height );
-    }
 
-    if( height > 1 && !query_yn( query_str, height ) ) {
-        return;
-    } else if( height == 1 ) {
+    {
         you.set_activity_level( ACTIVE_EXERCISE );
         weary_mult = 1.0f / you.exertion_adjusted_move_multiplier( ACTIVE_EXERCISE );
 
-        if( has_grapnel ) {
-            if( !query_yn( _( "Use your grappling hook to climb down?" ) ) ) {
+        std::string hint_remaining_fall;
+
+        if( height > 1 ) {
+            // After grappling or climbing down from a 2+ high ledge, the player will free-fall.
+            // We don't assess the expected damage but this simple warning is provided.
+            hint_remaining_fall = "\n";
+            hint_remaining_fall += string_format( n_gettext(
+                    "Even if you climb down safely, you will fall <color_yellow>at least %d story</color>.",
+                    "Even if you climb down safely, you will fall <color_red>at least %d stories</color>.",
+                    height - 1 ), height - 1 );
+        }
+
+        if( can_use_ladder ) {
+            if( !query_yn( _( "Use the ladder to climb down?" ) ) ) {
+                return;
+            } else {
+                has_grapnel = false;
+                web_rappel = false;
+            }
+        } else if( has_grapnel ) {
+            std::string query = _( "Use your grappling hook to climb down?" );
+            query += "%s";
+            if( !query_yn( query.c_str(), hint_remaining_fall ) ) {
                 has_grapnel = false;
             } else {
                 web_rappel = false;
             }
         }
 
-        if( !has_grapnel ) {
-            const char *query;
+        if( !can_use_ladder && !has_grapnel ) {
+            std::string query;
+            std::string hint_slip_risk;
+            std::string hint_fall_damage;
+            std::string hint_climb_back;
             if( web_rappel ) {
-                query = _( "Use your webs to descend?" );
+                query = string_format( n_gettext( "Use your webs to descend?",
+                                                  "Looks like %d stories.  Nothing your webs can't handle.  Descend?", height ), height );
             } else {
-                if( climb_cost <= 0 && fall_mod > 0.8 ) {
-                    query = _( "You probably won't be able to get up and jumping down may hurt.  Jump?" );
-                } else if( climb_cost <= 0 ) {
-                    query = _( "You probably won't be able to get back up.  Climb down?" );
+                // The most common query reads as follows:
+                //   It [seems somewhat risky] to climb down like this.
+                //   Falling [would hurt].
+                //   You [probably won't be able to climb back up].
+                query = "%s%s\n%s\n%s\n";
```

Comment:
Alternatives were to include a bunch of %s in one of the strings and use it as base (which is not an improvement) or to use std::string concatenation instead of printf-style formatting.  I don't use printf-like functions in my own work so I've got a weak grasp on them.  Open to suggestions.